be/src/io/cache/cache_lru_dumper.cpp

```diff
@@ -0,0 +1,375 @@
+    ::doris::io::cache::UInt128WrapperPb* hash_pb = entry->mutable_hash();
+    hash_pb->set_high(hash.high());
+    hash_pb->set_low(hash.low());
+    entry->set_offset(offset);
+    entry->set_size(size);
+
+    _current_dump_group_count++;
+    if (_current_dump_group_count >= 10000) {
+        RETURN_IF_ERROR(flush_current_group(out, filename));
+    }
+    return Status::OK();
+}
+
+Status CacheLRUDumper::flush_current_group(std::ofstream& out, std::string& filename) {
+    if (_current_dump_group_count == 0) {
+        return Status::OK();
+    }
+
+    // Record current position as group start offset
+    size_t group_start = out.tellp();
+
+    // Serialize and write the group
+    std::string serialized;
+    LOG(INFO) << "Before serialization: " << _current_dump_group.DebugString();
+    if (!_current_dump_group.SerializeToString(&serialized)) {
+        return Status::InternalError<false>("Failed to serialize LRUDumpEntryGroupPb");
+    }
+    LOG(INFO) << "Serialized size: " << serialized.size();
+    out.write(serialized.data(), serialized.size());
+    RETURN_IF_ERROR(check_ofstream_status(out, filename));
+
+    // Record group metadata
+    ::doris::io::cache::EntryGroupOffsetSizePb* group_info = _dump_meta.add_group_offset_size();
+    group_info->set_offset(group_start);
+    group_info->set_size(serialized.size());
+
+    // Reset for next group
+    _current_dump_group.Clear();
+    _current_dump_group_count = 0;
+    return Status::OK();
+}
+
+Status CacheLRUDumper::finalize_dump(std::ofstream& out, size_t entry_num,
+                                     std::string& tmp_filename, std::string& final_filename,
+                                     size_t& file_size) {
+    // Flush any remaining entries
+    if (_current_dump_group_count > 0) {
+        RETURN_IF_ERROR(flush_current_group(out, tmp_filename));
+    }
+
+    // Write meta information
+    _dump_meta.set_entry_num(entry_num);
+    size_t meta_offset = out.tellp();
+    LOG(INFO) << "dump meta: " << _dump_meta.DebugString();
+    std::string meta_serialized;
+    if (!_dump_meta.SerializeToString(&meta_serialized)) {
+        std::string warn_msg =
+                fmt::format("Failed to serialize LRUDumpMetaPb, file={}", tmp_filename);
+        LOG(WARNING) << warn_msg;
+        return Status::InternalError<false>("warn_msg");
+    }
+    out.write(meta_serialized.data(), meta_serialized.size());
+    RETURN_IF_ERROR(check_ofstream_status(out, tmp_filename));
+
+    // Write footer
+    Footer footer;
+    footer.meta_offset = htole64(meta_offset); // Explicitly convert to little-endian
+    footer.checksum = 0;                       // TODO: Calculate checksum
+    footer.version = 1;
+    std::memcpy(footer.magic, "DOR", 3);
+
+    out.write(reinterpret_cast<const char*>(&footer), sizeof(footer));
+    RETURN_IF_ERROR(check_ofstream_status(out, tmp_filename));
+
+    out.close();
```

Comment:
do we need a `sync`.
std::endl is a `sync` for stream, but it may introduce an `end of line`